File: data/alphafold/louvains.jl
#!/usr/bin/env julia
using Random
ROOT = readchomp(`git root`)


"Run a Cmd object, returning stdout, stderr, and exit code"
function execute(cmd::Cmd)
  out = Pipe()
  err = Pipe()

  process = run(pipeline(ignorestatus(cmd), stdout=out, stderr=err))
  close(out.in)
  close(err.in)

  (
    stdout = String(read(out)), 
    stderr = String(read(err)),  
    code = process.exitcode
  )
end




omes = vcat([readdir(d1; join=true) for d1 in readdir("PH"; join=true)]...)
println(length(omes), " proteomes")

omes = omes[.!isfile.(omes .* "/louvain.json.gz")]
println(length(omes), " todo")

shuffle!(omes)

for outdir in omes[1:min(1000,length(omes))]
    isfile(outdir*"/.inprogress") && continue
    touch(outdir*"/.inprogress")
    println(outdir)
    res = execute(`$ROOT/src/louvain.py $outdir/'AF*.json.gz' $outdir/louvain.json.gz`)
    if res.code == 0
        rm(outdir*"/.inprogress")
    else
        println("# STDOUT")
        println(res.stdout)
        println("# STDERR")
        println(res.stderr)
    end
end

File: data/alphafold/louvains.py
#!/usr/bin/env python3
import sys
sys.path.append("../../src")
import louvain
import os
from os.path import isfile
from pathlib import Path
from random import shuffle


d1s = os.listdir("PH")
omes = [os.path.join("PH", d1, d2) for d1 in d1s for d2 in os.listdir("PH/"+d1)]
print(len(omes), " proteomes")

omes = [ome for ome in omes if not isfile(ome + "/louvain.json.gz")]
print(len(omes), " todo")

shuffle(omes)

with open("louvains.log", "a") as log:
    for outdir in omes[0:min(1000,len(omes))]:
        if isfile(outdir+"/.inprogress"): continue
        Path(outdir+"/.inprogress").touch()
        print(outdir, file=log, flush=True)
        louvain.main(outdir+'/AF*.json.gz', outdir+'/louvain.json.gz', log=log)
        os.remove(outdir+"/.inprogress")

